build_automation/frontend/src/js/addTag.js

```diff
@@ -39,15 +48,41 @@ class TagCreation extends React.Component {
         var targetUrl = this.props.listUrl;
         const payload = { name: this.state.name, description: this.state.description };
         const currentInstance = this;
-        {
+        if (this.state.id > 0) {
+            // Update an exising Tag.
+            targetUrl = get_url(this.props.detailUrl, { id: this.state.id });
+            axios.patch(targetUrl, payload, {
+                responseType: 'json'
+            }).then(function (response) {
+                currentInstance.saveCallback(response.data, TAG_SAVE_TYPE.UPDATE);
+            }).catch(function (error) {
+                console.error("Error in updating the tag.", error);
+                console.error(error.response.data);
+            })
+        }
+        else {
             // Create a new Tag.
             axios.post(targetUrl, payload, {
                 responseType: 'json'
             }).then(function (response) {
                 currentInstance.saveCallback(response.data, TAG_SAVE_TYPE.CREATE);
+                currentInstance.setState({
+                    message: 'Save successful',
+                    messageType: 'info'});
             }).catch(function (error) {
                 console.error("Error in creating a new Tag", error);
+                console.error(error.response.data);
+                let errorMsg = 'Error in creating the Tag.';
+                if (!(JSON.stringify(error.response.data).indexOf('DUPLICATE_TAG_NAME') === -1)) {
```

Comment:
There is no error response with `DUPLICATE_TAG_NAME`. This `if` statement should be removed.